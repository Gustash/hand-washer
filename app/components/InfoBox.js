import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { colors } from 'app/config'

const InfoBox = ({ text, style, hidden = false }) => (
  <View style={[styles.container, { opacity: hidden ? 0 : 1 }, style]}>
    <Text style={styles.text}>{text}</Text>
  </View>
)

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.green,
    paddingHorizontal: 22,
    paddingVertical: 6,
    borderRadius: 4,
  },
  text: {
    color: colors.white,
    textAlign: 'center',
    fontSize: 17,
  },
})

export default InfoBox
