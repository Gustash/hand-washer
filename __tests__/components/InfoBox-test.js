import React from 'react'
import { View, Text } from 'react-native'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import InfoBox from '../../app/components/InfoBox'

const mockProps = props => ({
  text: '',
  style: null,
  ...props,
})

describe('<InfoBox />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = mockProps()
    wrapper = shallow(<InfoBox {...props} />)
  })

  it('matches snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('applies text', () => {
    expect(wrapper.find(Text).prop('children')).not.toBe(true)

    wrapper.setProps({
      ...props,
      text: 'Info Box Test',
    })

    expect(wrapper.find(Text).prop('children')).toBe('Info Box Test')
  })

  it('applies style', () => {
    expect(wrapper.find(View).prop('style')).toContain(null)

    wrapper.setProps({
      ...props,
      style: {
        marginTop: 1000,
      },
    })

    expect(wrapper.find(View).prop('style')).toContainEqual({
      marginTop: 1000,
    })
  })

  it('applies hidden', () => {
    expect(wrapper.find(View).prop('style')).toContainEqual({
      opacity: 1,
    })

    wrapper.setProps({
      ...props,
      hidden: true,
    })

    expect(wrapper.find(View).prop('style')).toContainEqual({
      opacity: 0,
    })
  })
})
