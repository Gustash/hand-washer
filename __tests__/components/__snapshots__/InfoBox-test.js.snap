// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox /> matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#00A778",
        "borderRadius": 4,
        "paddingHorizontal": 22,
        "paddingVertical": 6,
      },
      Object {
        "opacity": 1,
      },
      null,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
        "textAlign": "center",
      }
    }
  />
</View>
`;
